### PR TITLE
GHCR image push- tfc-bacckup-b2

### PR DESCRIPTION
**Pull request by Praveenraj-K**
-Pushing the images in both Docker Hub & GitHub Container Registry "GHCR.io"
-Only initiate when there is a commit in the main branch using the workflow trigger.

### CODE REVIEW

**Comment by devon-sil** [comment]
@Praveenraj-K Thanks for putting together a PR for this. I've added the developers for review.

**Review by hobbitronics — APPROVED** [review]
Looks good to me!

**Comment by Praveenraj-K** [comment]
Thanks Mike :)

File: .github/workflows/build-and-publish.yml [hunk 1]
```diff
@@ -17,19 +19,32 @@ jobs:
...
+            ${{ vars.DOCKER_ORG }}/${{ github.event.repository.name }}
+            ghcr.io/${{ github.repository }}
           tags: |
-            type=ref,event=branch            
-            type=semver,pattern={{version}}
-            type=semver,pattern={{major}}.{{minor}}
-            # set 'latest' tag for main branch
-            type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', 'main') }}
+            type=ref,event=branch  # Tag with branch name
+            type=raw,value=latest,enable=${{ github.ref == 'refs/heads/develop' }}  # Tag as 'latest' if it's the 'develop' branch
```
Comment (briskt) [inline .github/workflows/build-and-publish.yml]:
```suggestion
            type=raw,value=latest,enable=${{ github.ref == 'refs/heads/main' }}  # Tag as 'latest' if it's the 'main' branch
```
Looks like it was an intentional change from main to develop, but I think we want to keep latest pointing to main. Let me know if I missed something.

**Review by briskt — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-publish.yml (hunk 1 above)
Comment (briskt) [reply .github/workflows/build-and-publish.yml]: Or just remove `latest` altogether. I don't like the `latest` tag anyway.

File: .github/workflows/build-and-publish.yml [hunk 2]
```diff
@@ -2,10 +2,12 @@ name: Build and Publish
 
 on:
   push:
+    branches: [ 'develop' ]
+    tags: [ 'v*.*.*' ]
```
Comment (briskt) [inline .github/workflows/build-and-publish.yml]: If you want to restrict tag format, you need to use a regular expression like this:
```suggestion
    tags: [ 'v[0-9]+.[0-9]+.[0-9]+' ]
```
But better not to restrict it for now, I think.

Comment (briskt) [reply .github/workflows/build-and-publish.yml]: Hmm, I'm not 100% sure it fully supports regular expressions. Maybe just the character set (e.g. `[0-9]`) works. Here are examples: https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#patterns-to-match-branches-and-tags

Comment (briskt) [reply .github/workflows/build-and-publish.yml]: Yeah, if it did the `.` characters would have to be escaped. Oh well, like I said, better to just keep it simple for now.

**Pull request closed without merging**